### PR TITLE
#4277 Save multiple `quoteId` tokens in localstorage with `{website_code: quote_id}` pairs

### PR DESCRIPTION
**Pull request by neeeeecka**
**Related issue(s):**
* Fixes https://github.com/scandipwa/scandipwa/issues/4277

**Problem:**
* User's `quoteId` wasn't being updated properly when changing between websites' stores

**In this PR:**
* I modified `Token.js` to support multiple `quoteId` tokens from different websites based on `website_code`. 
* Modified `index.html` and `index.php` to create and set `window.website_code` global.

### CODE REVIEW

File: packages/scandipwa/public/index.html [hunk 1]
```diff
@@ -27,6 +27,8 @@
         // do reverse sort in order prevent an issue like store code `en` replaces store code `en_us`
         window.storeList = ['default'].sort().reverse();
         window.storeRegexText = `/(${window.storeList.join('|')})?`;
+
+        window.website_code = "";
```
Comment (zans-laksa) [inline packages/scandipwa/public/index.html:31]: Will it work if `website_code` is empty? Did you check it?
I'd suggest giving it the default "base" value instead.

File: packages/scandipwa/src/store/Cart/Cart.dispatcher.js [hunk 2]
```diff
@@ -368,7 +371,7 @@ export class CartDispatcher {
      * @param Dispatch dispatch
      * @return string quote id
      */
-    _getGuestQuoteId(dispatch) {
+    async _getGuestQuoteId(dispatch) {
```
Comment (zans-laksa) [inline packages/scandipwa/src/store/Cart/Cart.dispatcher.js:374]: There is no reason to make this `async`.

File: packages/scandipwa/src/store/Cart/Cart.dispatcher.js [hunk 3]
```diff
@@ -16,7 +16,11 @@ import { showNotification } from 'Store/Notification/Notification.action';
 import { getRegionIdOfRegionName } from 'Util/Address';
 import { getAuthorizationToken, isSignedIn } from 'Util/Auth';
 import { getGuestQuoteId, setGuestQuoteId } from 'Util/Cart';
-import { fetchMutation, fetchQuery, getErrorMessage } from 'Util/Request';
+import {
+    fetchMutation, fetchQuery, getErrorMessage
```
Comment (zans-laksa) [inline packages/scandipwa/src/store/Cart/Cart.dispatcher.js:20]: Please make it either one line or move each item to the separate line.

File: packages/scandipwa/src/util/Cart/Token.js [hunk 4]
```diff
@@ -13,28 +13,48 @@ import { isSignedIn } from 'Util/Auth';
...
 
 /** @namespace Util/Cart/Token/getGuestQuoteId */
 export const getGuestQuoteId = () => {
-    const {
-        token,
-        isCustomerToken
-    } = BrowserDatabase.getItem(GUEST_QUOTE_ID) || {};
+    const { website_code } = window;
+
+    const tokens = BrowserDatabase.getItem(WEBSITE_QUOTE_TOKENS);
```
Comment (zans-laksa) [inline packages/scandipwa/src/util/Cart/Token.js:34]: If you add a fallback, you can reduce nesting of the code below.
```javascript
const tokens = BrowserDatabase.getItem(WEBSITE_QUOTE_TOKENS) || {};
```

File: packages/scandipwa/src/util/Cart/Token.js [hunk 5]
```diff
@@ -13,28 +13,48 @@ import { isSignedIn } from 'Util/Auth';
 import BrowserDatabase from 'Util/BrowserDatabase';
 
 export const GUEST_QUOTE_ID = 'guest_quote_id';
+export const WEBSITE_QUOTE_TOKENS = 'Token.website_quote_tokens';
```
Comment (zans-laksa) [inline packages/scandipwa/src/util/Cart/Token.js:16]: 1. Why didn't you reuse the `GUEST_QUOTE_ID` variable?
2. Why did you leave it?
3. Why the values don't match the code style (lowercase letters divided by underscore)?

Please simply reuse the `GUEST_QUOTE_ID` value.

**Review by zans-laksa — CHANGES_REQUESTED** [review]

File: packages/scandipwa/public/index.html (hunk 1 above)
Comment (neeeeecka) [reply packages/scandipwa/public/index.html:31]: No it didn't work :d was empty string as key obviously, changed it to 'base'

**Review by zans-laksa — APPROVED** [review]

**Comment by carinadues** [comment]
Merging cuz can check only on PMREV.

**Pull request merged**